test/unit/limited_work_queue_test.rb

```diff
@@ -135,5 +135,179 @@ def merge(queue)
         assert started2 > ended0 || started2 > ended1, [ended0, ended1, started2].inspect
       end
     end
+
+    describe '#paused?' do
+      it 'should return false when @paused is false' do
+        @queue.instance_variable_set(:@paused, false)
+        assert_equal false, @queue.paused?
+      end
+
+      it 'should return true when @paused is true' do
+        @queue.instance_variable_set(:@paused, false)
+        assert_equal false, @queue.paused?
+      end
+    end
+
+    describe '#pause!' do
+      it 'should set @paused to true if @paused is currently false' do
+        assert_equal false, @queue.instance_variable_get('@paused')
+        @queue.pause!
+        assert_equal true, @queue.instance_variable_get('@paused')
+      end
+
+      it 'should set @paused to true if @paused is currently true' do
+        @queue.instance_variable_set(:@paused, true)
+        assert_equal true, @queue.instance_variable_get('@paused')
+        @queue.pause!
+        assert_equal true, @queue.instance_variable_get('@paused')
+      end
+    end
+
+    describe '#unpause!' do
+      it 'should set @paused to false if @paused is currently false' do
+        assert_equal false, @queue.instance_variable_get('@paused')
+        @queue.unpause!
+        assert_equal false, @queue.instance_variable_get('@paused')
+      end
+
+      it 'should set @paused to false if @paused is currently true' do
+        @queue.instance_variable_set(:@paused, true)
+        assert_equal true, @queue.instance_variable_get('@paused')
+        @queue.unpause!
+        assert_equal false, @queue.instance_variable_get('@paused')
+      end
+    end
+
+    describe '#add' do
```

Comment:
Makes sense, although I usually take that slightly more pragmatic route and fix things in place.  A method alias would have worked as well.